src/main/java/space/obminyashka/items_exchange/controller/UserController.java

```diff
@@ -126,6 +126,23 @@ public ResponseEntity<String> updateUserPassword(@Valid @RequestBody ChangePassw
         return new ResponseEntity<>(getMessageSource(ResponseMessagesHandler.PositiveMessage.CHANGED_USER_PASSWORD), HttpStatus.ACCEPTED);
     }
 
+    @PreAuthorize("hasAnyRole('USER', 'ADMIN', 'MODERATOR')")
```

Comment:
If we do reset in profile, it is ok but if we do reset like forgetting password, it is not ok.